lib/java/src/org/apache/thrift/transport/THttpClient.java

```diff
@@ -304,6 +304,9 @@ private void flushUsingHttpClient() throws TTransportException {
           throw new TTransportException(ioe);
         }
       }
+      if (post != null) {
+        post.releaseConnection();
```

Comment:
The code on line 295 appears to want to take care of this condition.  Perhaps other exceptions are occurring, or something else is happening like we're not actually consuming all the data that was sent?

From my read of this method "post" will never be null, so the check here and on line 294 are unnecessary.

I would check into the IOException handler above and see if it needs to be expanded, perhaps.